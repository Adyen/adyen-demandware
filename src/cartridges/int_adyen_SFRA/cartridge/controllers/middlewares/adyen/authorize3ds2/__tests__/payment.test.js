const handlePaymentsCall = require("../payment")

jest.mock('../errorHandler', () => ({
  handlePaymentError: jest.fn()
}))

jest.mock('../order')

let adyenCheckout;
let handlePaymentError;
let URLUtils;
let handlePlaceOrder;
let res;
let req;

beforeEach(() => {
  jest.clearAllMocks();
  res = { redirect: jest.fn() }
  adyenCheckout = require('*/cartridge/scripts/adyenCheckout');
  handlePaymentError = require('../errorHandler').handlePaymentError;
  URLUtils = require('dw/web/URLUtils');
  handlePlaceOrder = require("../order");
})
describe('Payment', () => {
  it('should handle invalid payment', () => {
    adyenCheckout.doPaymentDetailsCall.mockReturnValue({ error: true })
    handlePaymentsCall()
    expect(handlePaymentError).toBeCalledTimes(1)
  })
  it('should handle challengeShopper', () => {
    adyenCheckout.doPaymentDetailsCall.mockReturnValue({ resultCode: 'ChallengeShopper', action: 'mocked_action' })
    handlePaymentsCall({}, {}, {}, { res, next: jest.fn() })
    expect(URLUtils.url.mock.calls).toMatchSnapshot()
  })

  it('should place order', () => {
    adyenCheckout.doPaymentDetailsCall.mockReturnValue({ resultCode: 'Authorised', error: false })
    handlePaymentsCall({}, 'mocked_order', { custom: {} }, { res, next: jest.fn() })
    expect(handlePlaceOrder.mock.calls).toMatchSnapshot()
  })
})