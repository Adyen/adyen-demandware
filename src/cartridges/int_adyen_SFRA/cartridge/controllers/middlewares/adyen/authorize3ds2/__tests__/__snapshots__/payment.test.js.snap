// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment should handle challengeShopper 1`] = `
Array [
  Array [
    "Adyen-Adyen3DS2",
    "action",
    "\\"mocked_action\\"",
  ],
]
`;

exports[`Payment should place order 1`] = `
Array [
  Array [
    Object {
      "custom": Object {
        "adyenPaymentData": null,
      },
    },
    "mocked_order",
    Object {
      "error": false,
      "resultCode": "Authorised",
    },
    Object {
      "next": [MockFunction],
      "res": Object {
        "redirect": [MockFunction],
      },
    },
  ],
]
`;
